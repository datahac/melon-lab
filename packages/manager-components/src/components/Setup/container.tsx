import { withFormik } from 'formik';
import * as Yup from 'yup';
import Setup from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    name: Yup.string().required('name is required.'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values);
    }
  },
});

export default withFormValidation(Setup);
