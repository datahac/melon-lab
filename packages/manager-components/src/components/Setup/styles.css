.setup__info {
  margin-bottom: var(--space-l);
}

.setup__exchanges {
  margin-bottom: var(--space-l);
}

.setup__exchanges-checkbox {
  margin: var(--space-xxs) 0;
}

.setup__actions {
  display: flex;
  justify-content: space-between;
}

.setup__selected-exchange:not(:last-child)::after {
  content: ', ';
}
