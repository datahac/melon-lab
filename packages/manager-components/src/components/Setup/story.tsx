import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Setup from './index';

const data = {
  initialValues: {
    name: '',
  },
  config: {
    canonicalPriceFeedAddress: 'foo',
    competitionComplianceAddress: 'bar',
    onlyManagerCompetitionAddress: 'foo',
  },
  onSubmit: action('onSubmit'),
  signed: true,
  balances: {
    eth: 0.1
  }
};

storiesOf('Components|Setup', module).add('Default', () => {
  return <Setup {...data} />;
});
