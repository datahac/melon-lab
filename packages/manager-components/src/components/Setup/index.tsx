import React, { StatelessComponent, Fragment } from 'react';
import Button from '~/blocks/Button';
import Checkbox from '~/blocks/Checkbox';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';
import Modal from '~/blocks/Modal';
import Spinner from '~/blocks/Spinner';
import TermsAndConditions from '~/components/TermsAndConditions';
import Link from '~/blocks/Link';

import styles from './styles.css';
import Notification from '../../blocks/Notification/index';

interface FormValues {
  name: string;
}

export interface SetupProps {
  competitionName?: string;
  canonicalPriceFeedAddress: string;
  competitionComplianceAddress: string;
  noComplianceAddress: string;
  errors?: any;
  handleBlur?: () => void;
  handleChange?: () => void;
  handleSubmit?: () => void;
  isCompetition?: boolean;
  loading?: boolean;
  networkId?: string;
  touched?: any;
  values: FormValues;
  onClickAccept: () => void;
  onClickDecline: () => void;
  onClickConfirmFees: () => void;
  onClickDeclineFees: () => void;
  signed?: boolean;
  balances: {
    eth?: string;
    mln?: string;
  };
  address: string;
  fees?: any;
  FeeFormModal;
  FeeFormModalProps;
}

export const Setup: StatelessComponent<SetupProps> = ({
  competitionName = 'Naxos',
  canonicalPriceFeedAddress,
  competitionComplianceAddress,
  noComplianceAddress,
  errors,
  handleBlur,
  handleChange,
  handleSubmit,
  isCompetition,
  loading,
  networkId,
  touched,
  values,
  signed,
  onClickAccept,
  onClickDecline,
  balances,
  address,
  FeeFormModal,
  FeeFormModalProps = {},
}) => (
  <div className="setup">
    <style jsx>{styles}</style>

    <Modal
      title="Terms and Conditions"
      isOpen={!signed}
      PrimaryAction={Button}
      PrimaryActionProps={{
        children: 'Decline',
        style: 'secondary',
        onClick: onClickDecline,
      }}
      SecondaryAction={Button}
      SecondaryActionProps={{
        children: 'Accept',
        onClick: onClickAccept,
      }}
    >
      <TermsAndConditions />
    </Modal>

    <FeeFormModal {...FeeFormModalProps} />

    {loading ? (
      <Spinner
        icon
        size="small"
        text="Deploying your fund to the Ethereum blockchain..."
      />
    ) : (
      <Fragment>
        {balances && balances.eth === '0' ? (
          <Notification isWarning>
            <b>Insufficient ETH Balance</b>
            <p>
              You don't have enough Kovan Ether or Kovan W-ETH. Current
              balances: {balances.eth} ETH
              <br />
              To get started, head to our faucet to receive Kovan Ether and
              Kovan Melon
              <br />
              Once you have received ETH-T and MLN-T, go ahead and create your
              Melon fund.
            </p>

            <Link
              target="_blank"
              style="secondary"
              size="medium"
              href={`https://faucet.melon.fund/?address=${address}`}
            >
              Go to faucet
            </Link>
          </Notification>
        ) : (
          <Form onSubmit={handleSubmit}>
            <Input
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.name}
              required={true}
              name="name"
              type="text"
              placeholder="Fund name"
              error={touched.name && errors.name}
            />
            <h4>Melon Default Configuration:</h4>
            <p>
              For this version, the modules that your fund will use are
              predefined ie. you do not need to choose a module. For your
              record, below are the predefined modules for this version.
            </p>
            <div className="setup__exchanges">
              <h4>Exchange:</h4>
              <div className="setup__exchanges-checkbox">
                <Checkbox
                  name="exchanges"
                  value="OasisDex"
                  text="OasisDex"
                  disabled
                  defaultChecked
                />
              </div>
              <div className="setup__exchanges-checkbox">
                <Checkbox
                  name="exchanges"
                  value="0x relayers"
                  text="0x relayers"
                  disabled
                  defaultChecked
                />
              </div>
            </div>
            <div className="setup__info">
              Pricefeed:{' '}
              <a
                href={`https://${
                  networkId === '42' ? 'kovan.' : ''
                }etherscan.io/address/${canonicalPriceFeedAddress}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                <strong>Canonical PriceFeed</strong>
              </a>
              <br />
              Asset Registrar:{' '}
              <strong>Melon {competitionName} Asset Universe</strong>
              {isCompetition ? (
                <div>
                  Compliance (invest/redeem):{' '}
                  <a
                    href={`https://${
                      networkId === '42' ? 'kovan.' : ''
                    }etherscan.io/address/${competitionComplianceAddress}`}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <strong>
                      Only {competitionName} contribution contract can invest
                    </strong>
                  </a>
                </div>
              ) : (
                <div>
                  Compliance (invest/redeem):{' '}
                  <a
                    href={`https://${
                      networkId === '42' ? 'kovan.' : ''
                    }etherscan.io/address/${noComplianceAddress}`}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <strong>No compliance - anyone can invest (in WETH)</strong>
                  </a>
                </div>
              )}
              Risk Management: <strong>Disabled (all trades allowed)</strong>
            </div>
            <Button type="submit">Create and deploy my fund!</Button>
          </Form>
        )}
      </Fragment>
    )}
  </div>
);

export default Setup;
