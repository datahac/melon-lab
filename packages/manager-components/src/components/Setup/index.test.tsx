import React from 'react';
import Setup from './container';

const data = {
  initialValues: {
    name: '',
  },
  config: {
    canonicalPriceFeedAddress: 'foo',
    competitionComplianceAddress: 'bar',
    onlyManagerCompetitionAddress: 'foo',
  },
  onSubmit: () => null,
};

describe('Setup', () => {
  const defaultElement = <Setup {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
