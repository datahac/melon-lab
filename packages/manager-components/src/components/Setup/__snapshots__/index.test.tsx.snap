// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup should render correctly 1`] = `
<WithFormik(Function)
  config={
    Object {
      "canonicalPriceFeedAddress": "foo",
      "competitionComplianceAddress": "bar",
      "onlyManagerCompetitionAddress": "foo",
    }
  }
  initialValues={
    Object {
      "name": "",
    }
  }
  onSubmit={[MockFunction]}
>
  <Formik
    config={
      Object {
        "canonicalPriceFeedAddress": "foo",
        "competitionComplianceAddress": "bar",
        "onlyManagerCompetitionAddress": "foo",
      }
    }
    enableReinitialize={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "name": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <Component
      config={
        Object {
          "canonicalPriceFeedAddress": "foo",
          "competitionComplianceAddress": "bar",
          "onlyManagerCompetitionAddress": "foo",
        }
      }
      dirty={false}
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "name": "",
        }
      }
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      onSubmit={[MockFunction]}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "name": "",
        }
      }
    >
      <div
        className="jsx-__scopedHash setup"
      >
        <JSXStyle
          css="__scoped"
          styleId="__scopedHash"
        />
        <h3
          className="jsx-__scopedHash"
        >
          Setup your fund
        </h3>
        <Form
          onSubmit={[Function]}
        >
          <form
            className="form"
            onSubmit={[Function]}
          >
            <Input
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Fund name"
              required={true}
              type="text"
              value=""
            >
              <div
                className="jsx-__scopedHash input"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                <input
                  className="jsx-__scopedHash input__field"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Fund name"
                  type="text"
                  value=""
                />
                <div
                  className="jsx-__scopedHash input__error"
                />
              </div>
            </Input>
            <h4
              className="jsx-__scopedHash"
            >
              Melon Default Configuration:
            </h4>
            <p
              className="jsx-__scopedHash"
            >
              For this version, the modules that your fund will use are predefined ie. you do not need to choose a module. For your record, below are the predefined modules for this version.
            </p>
            <div
              className="jsx-__scopedHash setup__exchanges"
            >
              <h4
                className="jsx-__scopedHash"
              >
                Exchange:
              </h4>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="OasisDex"
                  value="OasisDex"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="OasisDex"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      OasisDex
                    </span>
                  </label>
                </Checkbox>
              </div>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="0x relayers"
                  value="0x relayers"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="0x relayers"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      0x relayers
                    </span>
                  </label>
                </Checkbox>
              </div>
            </div>
            <div
              className="jsx-__scopedHash setup__info"
            >
              Pricefeed:
               
              <a
                className="jsx-__scopedHash"
                href="https://etherscan.io/address/foo"
                rel="noopener noreferrer"
                target="_blank"
              >
                <strong
                  className="jsx-__scopedHash"
                >
                  Canonical PriceFeed
                </strong>
              </a>
              <br
                className="jsx-__scopedHash"
              />
              Asset Registrar: 
              <strong
                className="jsx-__scopedHash"
              >
                Melon 
                Naxos
                 Asset Universe
              </strong>
              <div
                className="jsx-__scopedHash"
              >
                Compliance (invest/redeem):
                 
                <a
                  className="jsx-__scopedHash"
                  href="https://etherscan.io/address/foo"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <strong
                    className="jsx-__scopedHash"
                  >
                    Only manager can invest (in WETH)
                  </strong>
                </a>
              </div>
              Risk Management: 
              <strong
                className="jsx-__scopedHash"
              >
                Disabled (all trades allowed)
              </strong>
            </div>
            <Button
              type="submit"
            >
              <button
                className="jsx-__scopedHash button button--medium button--primary"
                onClick={[Function]}
                type="submit"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                Create and deploy my fund!
              </button>
            </Button>
          </form>
        </Form>
      </div>
    </Component>
  </Formik>
</WithFormik(Function)>
`;

exports[`Setup should render correctly with isCompetition 1`] = `
<WithFormik(Function)
  config={
    Object {
      "canonicalPriceFeedAddress": "foo",
      "competitionComplianceAddress": "bar",
      "onlyManagerCompetitionAddress": "foo",
    }
  }
  initialValues={
    Object {
      "name": "",
    }
  }
  isCompetition={true}
  onSubmit={[MockFunction]}
>
  <Formik
    config={
      Object {
        "canonicalPriceFeedAddress": "foo",
        "competitionComplianceAddress": "bar",
        "onlyManagerCompetitionAddress": "foo",
      }
    }
    enableReinitialize={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "name": "",
      }
    }
    isCompetition={true}
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <Component
      config={
        Object {
          "canonicalPriceFeedAddress": "foo",
          "competitionComplianceAddress": "bar",
          "onlyManagerCompetitionAddress": "foo",
        }
      }
      dirty={false}
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "name": "",
        }
      }
      isCompetition={true}
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      onSubmit={[MockFunction]}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "name": "",
        }
      }
    >
      <div
        className="jsx-__scopedHash setup"
      >
        <JSXStyle
          css="__scoped"
          styleId="__scopedHash"
        />
        <h3
          className="jsx-__scopedHash"
        >
          Setup your fund
        </h3>
        <Form
          onSubmit={[Function]}
        >
          <form
            className="form"
            onSubmit={[Function]}
          >
            <Input
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Fund name"
              required={true}
              type="text"
              value=""
            >
              <div
                className="jsx-__scopedHash input"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                <input
                  className="jsx-__scopedHash input__field"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Fund name"
                  type="text"
                  value=""
                />
                <div
                  className="jsx-__scopedHash input__error"
                />
              </div>
            </Input>
            <h4
              className="jsx-__scopedHash"
            >
              Melon Default Configuration:
            </h4>
            <p
              className="jsx-__scopedHash"
            >
              For this version, the modules that your fund will use are predefined ie. you do not need to choose a module. For your record, below are the predefined modules for this version.
            </p>
            <div
              className="jsx-__scopedHash setup__exchanges"
            >
              <h4
                className="jsx-__scopedHash"
              >
                Exchange:
              </h4>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="OasisDex"
                  value="OasisDex"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="OasisDex"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      OasisDex
                    </span>
                  </label>
                </Checkbox>
              </div>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="0x relayers"
                  value="0x relayers"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="0x relayers"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      0x relayers
                    </span>
                  </label>
                </Checkbox>
              </div>
            </div>
            <div
              className="jsx-__scopedHash setup__info"
            >
              Pricefeed:
               
              <a
                className="jsx-__scopedHash"
                href="https://etherscan.io/address/foo"
                rel="noopener noreferrer"
                target="_blank"
              >
                <strong
                  className="jsx-__scopedHash"
                >
                  Canonical PriceFeed
                </strong>
              </a>
              <br
                className="jsx-__scopedHash"
              />
              Asset Registrar: 
              <strong
                className="jsx-__scopedHash"
              >
                Melon 
                Naxos
                 Asset Universe
              </strong>
              <div
                className="jsx-__scopedHash"
              >
                Compliance (invest/redeem):
                 
                <a
                  className="jsx-__scopedHash"
                  href="https://etherscan.io/address/bar"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <strong
                    className="jsx-__scopedHash"
                  >
                    Only 
                    Naxos
                     contribution contract can invest
                  </strong>
                </a>
              </div>
              Risk Management: 
              <strong
                className="jsx-__scopedHash"
              >
                Disabled (all trades allowed)
              </strong>
            </div>
            <Button
              type="submit"
            >
              <button
                className="jsx-__scopedHash button button--medium button--primary"
                onClick={[Function]}
                type="submit"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                Create and deploy my fund!
              </button>
            </Button>
          </form>
        </Form>
      </div>
    </Component>
  </Formik>
</WithFormik(Function)>
`;

exports[`Setup should render correctly with loading 1`] = `
<WithFormik(Function)
  config={
    Object {
      "canonicalPriceFeedAddress": "foo",
      "competitionComplianceAddress": "bar",
      "onlyManagerCompetitionAddress": "foo",
    }
  }
  initialValues={
    Object {
      "name": "",
    }
  }
  loading={true}
  onSubmit={[MockFunction]}
>
  <Formik
    config={
      Object {
        "canonicalPriceFeedAddress": "foo",
        "competitionComplianceAddress": "bar",
        "onlyManagerCompetitionAddress": "foo",
      }
    }
    enableReinitialize={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "name": "",
      }
    }
    isInitialValid={false}
    loading={true}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <Component
      config={
        Object {
          "canonicalPriceFeedAddress": "foo",
          "competitionComplianceAddress": "bar",
          "onlyManagerCompetitionAddress": "foo",
        }
      }
      dirty={false}
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "name": "",
        }
      }
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      loading={true}
      onSubmit={[MockFunction]}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "name": "",
        }
      }
    >
      <div
        className="jsx-__scopedHash setup"
      >
        <JSXStyle
          css="__scoped"
          styleId="__scopedHash"
        />
        <h3
          className="jsx-__scopedHash"
        >
          Setup your fund
        </h3>
        <p
          className="jsx-__scopedHash"
        >
          Deploying your fund to the Ethereum blockchain
        </p>
        <Form
          onSubmit={[Function]}
        >
          <form
            className="form"
            onSubmit={[Function]}
          >
            <Input
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Fund name"
              required={true}
              type="text"
              value=""
            >
              <div
                className="jsx-__scopedHash input"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                <input
                  className="jsx-__scopedHash input__field"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Fund name"
                  type="text"
                  value=""
                />
                <div
                  className="jsx-__scopedHash input__error"
                />
              </div>
            </Input>
            <h4
              className="jsx-__scopedHash"
            >
              Melon Default Configuration:
            </h4>
            <p
              className="jsx-__scopedHash"
            >
              For this version, the modules that your fund will use are predefined ie. you do not need to choose a module. For your record, below are the predefined modules for this version.
            </p>
            <div
              className="jsx-__scopedHash setup__exchanges"
            >
              <h4
                className="jsx-__scopedHash"
              >
                Exchange:
              </h4>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="OasisDex"
                  value="OasisDex"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="OasisDex"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      OasisDex
                    </span>
                  </label>
                </Checkbox>
              </div>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="0x relayers"
                  value="0x relayers"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="0x relayers"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      0x relayers
                    </span>
                  </label>
                </Checkbox>
              </div>
            </div>
            <div
              className="jsx-__scopedHash setup__info"
            >
              Pricefeed:
               
              <a
                className="jsx-__scopedHash"
                href="https://etherscan.io/address/foo"
                rel="noopener noreferrer"
                target="_blank"
              >
                <strong
                  className="jsx-__scopedHash"
                >
                  Canonical PriceFeed
                </strong>
              </a>
              <br
                className="jsx-__scopedHash"
              />
              Asset Registrar: 
              <strong
                className="jsx-__scopedHash"
              >
                Melon 
                Naxos
                 Asset Universe
              </strong>
              <div
                className="jsx-__scopedHash"
              >
                Compliance (invest/redeem):
                 
                <a
                  className="jsx-__scopedHash"
                  href="https://etherscan.io/address/foo"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <strong
                    className="jsx-__scopedHash"
                  >
                    Only manager can invest (in WETH)
                  </strong>
                </a>
              </div>
              Risk Management: 
              <strong
                className="jsx-__scopedHash"
              >
                Disabled (all trades allowed)
              </strong>
            </div>
            <Button
              type="submit"
            >
              <button
                className="jsx-__scopedHash button button--medium button--primary"
                onClick={[Function]}
                type="submit"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                Create and deploy my fund!
              </button>
            </Button>
          </form>
        </Form>
      </div>
    </Component>
  </Formik>
</WithFormik(Function)>
`;

exports[`Setup should render correctly with networkId 1`] = `
<WithFormik(Function)
  config={
    Object {
      "canonicalPriceFeedAddress": "foo",
      "competitionComplianceAddress": "bar",
      "onlyManagerCompetitionAddress": "foo",
    }
  }
  initialValues={
    Object {
      "name": "",
    }
  }
  networkId="42"
  onSubmit={[MockFunction]}
>
  <Formik
    config={
      Object {
        "canonicalPriceFeedAddress": "foo",
        "competitionComplianceAddress": "bar",
        "onlyManagerCompetitionAddress": "foo",
      }
    }
    enableReinitialize={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "name": "",
      }
    }
    isInitialValid={false}
    networkId="42"
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <Component
      config={
        Object {
          "canonicalPriceFeedAddress": "foo",
          "competitionComplianceAddress": "bar",
          "onlyManagerCompetitionAddress": "foo",
        }
      }
      dirty={false}
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "name": "",
        }
      }
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      networkId="42"
      onSubmit={[MockFunction]}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "name": "",
        }
      }
    >
      <div
        className="jsx-__scopedHash setup"
      >
        <JSXStyle
          css="__scoped"
          styleId="__scopedHash"
        />
        <h3
          className="jsx-__scopedHash"
        >
          Setup your fund
        </h3>
        <Form
          onSubmit={[Function]}
        >
          <form
            className="form"
            onSubmit={[Function]}
          >
            <Input
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Fund name"
              required={true}
              type="text"
              value=""
            >
              <div
                className="jsx-__scopedHash input"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                <input
                  className="jsx-__scopedHash input__field"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Fund name"
                  type="text"
                  value=""
                />
                <div
                  className="jsx-__scopedHash input__error"
                />
              </div>
            </Input>
            <h4
              className="jsx-__scopedHash"
            >
              Melon Default Configuration:
            </h4>
            <p
              className="jsx-__scopedHash"
            >
              For this version, the modules that your fund will use are predefined ie. you do not need to choose a module. For your record, below are the predefined modules for this version.
            </p>
            <div
              className="jsx-__scopedHash setup__exchanges"
            >
              <h4
                className="jsx-__scopedHash"
              >
                Exchange:
              </h4>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="OasisDex"
                  value="OasisDex"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="OasisDex"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      OasisDex
                    </span>
                  </label>
                </Checkbox>
              </div>
              <div
                className="jsx-__scopedHash setup__exchanges-checkbox"
              >
                <Checkbox
                  defaultChecked={true}
                  disabled={true}
                  name="exchanges"
                  text="0x relayers"
                  value="0x relayers"
                >
                  <label
                    className="jsx-__scopedHash checkbox"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    <input
                      className="jsx-__scopedHash checkbox__input"
                      defaultChecked={true}
                      disabled={true}
                      name="exchanges"
                      type="checkbox"
                      value="0x relayers"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__checkmark"
                    />
                    <span
                      className="jsx-__scopedHash checkbox__text"
                    >
                      0x relayers
                    </span>
                  </label>
                </Checkbox>
              </div>
            </div>
            <div
              className="jsx-__scopedHash setup__info"
            >
              Pricefeed:
               
              <a
                className="jsx-__scopedHash"
                href="https://kovan.etherscan.io/address/foo"
                rel="noopener noreferrer"
                target="_blank"
              >
                <strong
                  className="jsx-__scopedHash"
                >
                  Canonical PriceFeed
                </strong>
              </a>
              <br
                className="jsx-__scopedHash"
              />
              Asset Registrar: 
              <strong
                className="jsx-__scopedHash"
              >
                Melon 
                Naxos
                 Asset Universe
              </strong>
              <div
                className="jsx-__scopedHash"
              >
                Compliance (invest/redeem):
                 
                <a
                  className="jsx-__scopedHash"
                  href="https://kovan.etherscan.io/address/foo"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <strong
                    className="jsx-__scopedHash"
                  >
                    Only manager can invest (in WETH)
                  </strong>
                </a>
              </div>
              Risk Management: 
              <strong
                className="jsx-__scopedHash"
              >
                Disabled (all trades allowed)
              </strong>
            </div>
            <Button
              type="submit"
            >
              <button
                className="jsx-__scopedHash button button--medium button--primary"
                onClick={[Function]}
                type="submit"
              >
                <JSXStyle
                  css="__scoped"
                  styleId="__scopedHash"
                />
                Create and deploy my fund!
              </button>
            </Button>
          </form>
        </Form>
      </div>
    </Component>
  </Formik>
</WithFormik(Function)>
`;
